pallets/delegation/src/lib.rs

```diff
@@ -939,3 +943,42 @@ impl<T: Config> Pallet<T> {
 		Ok((removals, consumed_weight))
 	}
 }
+
+#[derive(Clone, Debug, Encode, Decode, PartialEq, Eq, TypeInfo)]
+pub struct DelegationAc<T: Config>(DelegationNodeIdOf<T>, u32);
+impl<T: Config> attestation::AttestationAccessControl<DelegatorIdOf<T>, DelegationNodeIdOf<T>> for DelegationAc<T> {
+	fn can_attest(&self, who: &DelegatorIdOf<T>) -> Result<Weight, DispatchError> {
+		match Pallet::<T>::is_delegating(who, &self.0, self.1)? {
+			(true, checks) => Ok(<T as Config>::WeightInfo::can_attest(checks)),
+			_ => Err(Error::<T>::AccessDenied.into()),
+		}
+	}
+
+	fn can_revoke(&self, who: &DelegatorIdOf<T>, auth_id: &DelegationNodeIdOf<T>) -> Result<Weight, DispatchError> {
+		ensure!(auth_id == &self.0, Error::<T>::AccessDenied);
+
+		match Pallet::<T>::is_delegating(who, &self.0, self.1)? {
+			(true, checks) => Ok(<T as Config>::WeightInfo::can_revoke(checks)),
+			_ => Err(Error::<T>::AccessDenied.into()),
+		}
+	}
+
+	fn can_remove(&self, who: &DelegatorIdOf<T>, auth_id: &DelegationNodeIdOf<T>) -> Result<Weight, DispatchError> {
+		ensure!(auth_id == &self.0, Error::<T>::AccessDenied);
+
+		match Pallet::<T>::is_delegating(who, &self.0, self.1)? {
```

Comment:
This check seems to be incorrect for removal access checking because it returns `(false, _)` when the delegation is revoked. Also, we once said we don't allow parental nodes to remove children except if thei remove their own node. 

Shouldn't this suffice?
```suggestion
		let delegation_node = <DelegationNodes<T>>::get(&self.0).ok_or(Error::<T>::DelegationNotFound)?; 
		if &who == &delegation_node.details.owner { ... } else { Err(...) }
```

Reply:
> Also, we once said we don't allow parental nodes to remove children except if their remove their own node.

This is not about removing delegations but attestations. All parent nodes can remove the attestation?